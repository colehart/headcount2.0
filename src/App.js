import React, { Component } from 'react';
import Header from './Header/Header'
import CardContainer from './CardContainer/CardContainer'
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <Header />
        <CardContainer />
      </div>
    );
  }
}

export default App;
