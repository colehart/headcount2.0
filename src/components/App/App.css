/*APP COMPONENT RELATED CSS GOES HERE*/
.App {
  height: 100vh;
  width: 100vw;
  margin: 0 auto;
  padding: 0;
  display: grid;
  grid-template-rows: 25% 1fr;
  grid-template-columns: 1fr;
}