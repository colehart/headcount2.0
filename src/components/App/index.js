import React, { Component } from 'react';
import Header from '../Header'
import CardContainer from '../CardContainer'
import DistrictRepository from '../../helper.js';
import kinderData from '../../data/kindergartners_in_full_day_program'
import './App.css';

class App extends Component {
  constructor() {
    super()
    this.state = {
      kinder: {}
    }
  }

  findAllMatches = (query) => {
    const kinder = new DistrictRepository(kinderData)
    const matchingNames = kinder.findAllMatches(query)
    console.log(`query: ${query}, matchingName: ${matchingNames}`)
    matchingNames.forEach(name => {
      const district = kinder.findByName(name)
      this.setState({kinder: district})
    })

  }

  componentDidMount() {
    const kinder = new DistrictRepository(kinderData)

    this.setState({
      kinder: kinder.stats
    })
  }

  render() {
    const kinder = this.state.kinder

    return (
      <div className="App">
        <Header findAllMatches={this.findAllMatches}/>
        <CardContainer districts={kinder} />
      </div>
    );
  }
}

export default App