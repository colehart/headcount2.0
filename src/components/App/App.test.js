import React from 'react';
import { shallow, mount } from 'enzyme';
import App from './index';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />)
  })

  it('should render all components', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should have initial state of all kindergarten districts', () => {
    wrapper = mount(<App />)

    expect(Object.keys(wrapper.state().kinder).length).toBe(181)
  })

  // it('should find all matches and update state', () => {
  //   const initialState = {'COLORADO': {}, 'ALAMOSA': {}, 'COLLIN COUNTY': {} }
  //   const mockQuery = 'Co'
  //   const expected = {'COLORADO': {}, 'COLLIN COUNTY': {} }

  //   wrapper.setState({ kinder: initialState })
  //   wrapper.instance().findAllMatches(mockQuery)

  //   expect(wrapper.state('kinder')).toEqual(expected)
  // })
})
