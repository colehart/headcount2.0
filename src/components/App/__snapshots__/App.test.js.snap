// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render all components 1`] = `
<div
  className="App"
>
  <Header />
  <CardContainer />
</div>
`;
